### ISSUE TITLE
Does it Work

### ISSUE THREAD

**Issue by 578123043** [facebookresearch/SpanBERT#36]

**Comment by 578123043** [comment]
![image](https://user-images.githubusercontent.com/16147509/71884198-cd61e880-3172-11ea-9796-27f9e881f133.png)

In FP16 , output is nan , and in mid layer(maybe 17th or 18th) in the bert , I found that the attention is -3000，and softmax result is nan

**Comment by mandarjoshi90** [comment]
I will need more information to help you debug. Could you please include what command you're running?

**Comment by 578123043** [comment]
YES ， I use {https://dl.fbaipublicfiles.com/fairseq/models/spanbert_hf.tar.gz} to runing Huggingface`s run_squad , it Failed
Does it work with the latest amp？

**Comment by mandarjoshi90** [comment]
We haven't tested this with the new HF code. ICYMI, there's a run_squad.py in this repo. I'd recommend using that.

**Comment by marcos0318** [comment]
When I tried to finetune spanbert large on my own task using fp16 with the "amp" module from apex, I meet a similar error. 

I am using the code from huggingface and tried both using "Spanbert/spanbert-large-cased" and the model binaries provided in this repo. Both gives the identical result, which kept telling me gradient overflow and rescaling the loss. 

Interestingly, when I tried to replace spanbert-large with BERT base/large, and spanbert-base model, these models work perfectly and achieved expected results. 

Also, the spanbert-large work very well when I turn off the fp16 training. 

Here I found this guy cannot run fp16 with spanbert on another task. 

As a conclusion, I guess that the spanbert-large model does work alone with the new nvidia amp, but I am not sure about it. 

This could be an problem when using the model to solve coreference problems, because runing coreference models requires large single gpu memory.

**Comment by YuxianMeng** [comment]
Met same issue here, is there any solution yet?
